File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.py
def solution(storey):
    answer = 0
    
    while storey > 0:
        number = storey % 10
        storey //= 10
        
        if number > 5:
            answer += (10 - number)
            storey += 1
        elif number < 5:
            answer += number
        else:
            if storey % 10 >= 5:
                answer += 5
                storey += 1
            else:
                answer += 5
                
    return answer

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.java
class Solution {
    public int solution(int storey) {
        int answer = 0;
        while (storey > 0) {
            int number = storey % 10;
            storey = storey / 10;
            
            if (number > 5){
                answer += (10 - number);
                storey += 1;
            } else if (number < 5){
                answer += number;
            } else if (storey % 10 >= 5){
                answer += 5;
                storey++;
            } else {
                answer += 5;
            }
        }
        
        return answer;
    }
}